lib/rsync_store.ml

```diff
@@ -101,7 +101,7 @@ let build t ?base ~id fn =
       (fun r ->
       begin match r with
           | Ok () -> Rsync.rename_with_sharing ~mode:t.mode ~base ~src:result_tmp ~dst:result
-          | Error _ -> Lwt.return_unit
+          | Error _ -> Rsync.delete result_tmp >>= fun () -> Lwt.return_unit
```

Comment:
```suggestion
          | Error _ -> Rsync.delete result_tmp
```

Reply:
Thanks - I have incorporated this in the next commit